au/code/au/utility/test/type_traits_test.cc

```diff
@@ -75,5 +75,16 @@ TEST(IncludeInPackIf, MakesPackOfEverythingThatMatches) {
         Pack<uint8_t, uint64_t>>();
 }
 
+TEST(CommonTypeButPreserveIntSignedness, CommonTypeIfItIsNotIntegral) {
+    StaticAssertTypeEq<CommonTypeButPreserveIntSignedness<int, double>, double>();
+    StaticAssertTypeEq<CommonTypeButPreserveIntSignedness<float, int>, float>();
+    StaticAssertTypeEq<CommonTypeButPreserveIntSignedness<double, float>, double>();
+}
```

Comment:
Should we add some unsigned tests to this test case?

Reply:
`u` got it

Reply:
🤦‍♂️